internal/k8s/controller.go

```diff
@@ -1504,10 +1504,26 @@ func (lbc *LoadBalancerController) createVirtualServer(virtualServer *conf_v1alp
 	}
 
 	endpoints := make(map[string][]string)
+	externalNameSvcs := make(map[string]bool)
 
 	for _, u := range virtualServer.Spec.Upstreams {
-		endpointsKey := configs.GenerateEndpointsKey(virtualServer.Namespace, u.Service, u.Port)
-		endpoints[endpointsKey] = lbc.getEndpointsForService(virtualServer.Namespace, u.Service, int(u.Port))
+		backend := &extensions.IngressBackend{
```

Comment:
seems like you forgot to update this part or vice versa.  the logic in the  loop `for _, u := range vsr.Spec.Upstreams` in line 1569 looks different to this one.